File: 2/2_cuatrimestre/PDOO/practicas/practica 3/examen2_p3_ruby/SuppliesPackage.rb
module Deepspace

    class SuppliesPackage
        
        def initialize (ammoPower, fuelUnits, shieldPower)
            @ammoPower = ammoPower
            @fuelUnits = fuelUnits
            @shieldPower = shieldPower
        end

        def self.newCopy (s)
            new(s.ammoPower, s.fuelUnits, s.shieldPower)
        end


        def ammoPower
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end
        
        def shieldPower
            @shieldPower
        end
        
        def to_s
            "AmmoPower: #{@ammoPower}, FuelUnits: #{@fuelUnits}, ShieldPower: #{@shieldPower}"
        end
    end

end

File: 2/2_cuatrimestre/PDOO/practicas/practica 1/deepSpace/src/deepspace/SuppliesPackage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author manbolq
 */
public class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    
    
    /**************************************************************************/
    /*************************    CONSTRUCTORES   *****************************/
    /**************************************************************************/
    
    SuppliesPackage(float ammoPower, float fuelUnits, float shieldPower){
        this.ammoPower = ammoPower;
        this.fuelUnits = fuelUnits;
        this.shieldPower = shieldPower;
    }
    
    SuppliesPackage(SuppliesPackage s){
        ammoPower = s.ammoPower;
        fuelUnits = s.fuelUnits;
        shieldPower = s.shieldPower;
    }
    
    /**************************************************************************/
    /**************************    CONSULTORES   ******************************/
    /**************************************************************************/
    
    float getAmmoPower(){
        return ammoPower;
    }
    
    float getFuelUnits(){
        return fuelUnits;
    }
    
    float getShieldPower(){
        return shieldPower;
    }
    
}
